nni/experiment/pipe.py

```diff
@@ -52,7 +52,7 @@ def __init__(self, experiment_id: str):
 
         def connect(self) -> BufferedIOBase:
             conn, _ = self._socket.accept()
-            self.file = conn.makefile('w+b')
+            self.file = conn.makefile('rwb')
```

Comment:
Why is this?

Reply:
https://github.com/microsoft/nni/pull/3183 @liuzhe-lz can answer this question